contrib/babelfishpg_tds/src/backend/tds/tdsrpc.c

```diff
@@ -1787,6 +1787,49 @@ ReadParameters(TDSRequestSP request, uint64_t offset, StringInfo message, int *p
 					CheckPLPStatusNotOK(temp, retStatus);
 				}
 				break;
+			case TDS_TYPE_CLRUDT:
+				{
+					uint16_t	len;
+					uint8_t   typenamelen;
+					StringInfoData   typeName;
+
+					initStringInfo(&typeName);
+
+					memcpy(&len, &message->data[offset], sizeof(len));
+					offset += sizeof(len);
+
+					temp->maxLen = len;
+
+					/* We are reading the type name for spatial datatypes (geometry or geography) */
+					memcpy(&typenamelen, &message->data[offset], sizeof(typenamelen));
+					offset += sizeof(typenamelen);
+
+					TdsUTF16toUTF8StringInfo(&typeName, &(message->data[offset]), 2*typenamelen);
+					offset += 2*typenamelen;
+
+					if(!(pg_strcasecmp(typeName.data, "geometry") == 0 || pg_strcasecmp(typeName.data, "geography") == 0)){
+						ereport(ERROR,
+							(errcode(ERRCODE_PROTOCOL_VIOLATION),
+							errmsg("The incoming tabular data stream (TDS) remote procedure call (RPC) protocol stream is incorrect. "
+									"UdtTypeName is incorrect.")));
+					}
+
+					/* we set the column metadata for the appropriate types and also set TDS type from CLRUDT type to respective geometry or geography type */
+					if(pg_strcasecmp(typeName.data, "geometry") == 0){
```

Comment:
Could you try running the basic tests with Valgrind as well? Ping me if you have any questions about doing that, it's pretty straightforward. Just as a sanity check for memory safety

Reply:
Yeah, I ran tests with valgrind, and didn't see any memory issues, So, it looks fine